### ISSUE TITLE
Add AWS Accounts with different role path

### ISSUE THREAD

**Issue by Yivensky** [awslabs/sandbox-accounts-for-events#82]
Hello,
It is hardcoded that the role must be in the path / but for security reasons I add high permissive role in another path and restricted them by using SCP.
However, the interface doesn't allow me to add the AWS Account because he don't find the role, because it's not in the path /

I know where I can change the code (change regex in the input and allow /, and type for exemple mypath/myrolename it would not break the code) but I am pretty bad with make and react etc...to update the deployment

My question is : In the future, for next releases, how can I update my deployment  with new version of this tools and keep my settings?
i tried to make deploy again to get new .zip, and create a new deployment in my amplify app and use amplify.zip, I broke the webpage

THank you

**Comment by moellr** [comment]
First of all: Good idea to allow non-"/"-path roles in the register account dialog. Will add it in one of the next releases.

Regarding updates: There is no safe update path of an existing deployment. The only way would be to update the Amplify code in your repo, zip & upload it to S3 and re-execute the CodeBuild project for Amplify deployment. But this will be at your own risk, it is not a documented/safe update path.

**Comment by moellr** [comment]
Regarding IAM roles including "/": The change has just gone live with v1.3.1, RegisterAccount dialog now supports IAM roles with custom paths as well.

**Comment by Yivensky** [comment]
Thank you for the update
What is the IAM role used by the app to assume the role?
Registering account xxxxxx.....Error trying to assume IAM role /xxxx/role in account xxxxx.

Is probably because my SCP protect it but I can't figure out what is its role (and I would like to avoid allowing root account in scp)

**Comment by moellr** [comment]
I did some deep dive today. Outcomes:

* The IAM role in the child accounts is assumed by a number of resources, mainly Lambdas across this "Sandbox Account for Events" project and the underlying DCE project. If you need to list all possible principles, I recommend using CloudTrail to capture all AssumeRole events to compile a list of all IAM principles that assume this IAM role during the account lifecycle. (I do not know enough details about the DCE project to be able to provide that list).

* Sad news: The underlying DCE project doesn't seem to support IAM roles with paths. Registration works (after I adjusted the ```safeAdminApi-main``` backend Lambda in this "Sandbox Accounts for Events" project, which caused your issue mentioned above), but it ultimately fails when the DCE project is trying to execute the AWS Nuke CodeBuild job. The ```Lambda process_reset_queue-dce``` does not pass the IAM role name correctly to AWS Nuke, which causes the CodeBuild execution to fail. I tried to dig into the DCE code, and this line seems to be the root cause for it, as it only passes IAM role name characters behind the last ```/```: https://github.com/Optum/dce/blob/71356bace13df0ab330c298d98ddd1b2add68972/pkg/arn/arn.go#L80C10-L80C22

In short: IAM roles with paths are not supported by DCE as of now, and will therefor not work within the scope of the project. I will adjust the ```safeAdminApi-main``` backend Lambda to handle IAM roles with paths properly in one of the next releases, but it won't add any value unless the DCE project supports it at some point in future.

**Comment by Yivensky** [comment]
Thank you for the investigation and for contributing to this tools which is very useful
